automation/jleechanorg_pr_automation/automation_utils.py

```
@@ -210,65 +210,61 @@ def execute_subprocess_with_timeout(cls, command: list, timeout: int = None,
-            with open(file_path) as f:
-                fcntl.flock(f.fileno(), fcntl.LOCK_SH)  # Shared lock for reading
-                data = json.load(f)
-                fcntl.flock(f.fileno(), fcntl.LOCK_UN)  # Unlock
-                return data
-        except (FileNotFoundError, json.JSONDecodeError):
-            return {}
+        """Safely read JSON file with file locking (delegates to json_manager)"""
+        return json_manager.read_json(str(file_path), {})
 
     @classmethod
     def safe_write_json(cls, file_path: Path, data: dict):
```

Ruff (0.14.10)

Missing return type annotation for classmethod `safe_write_json`

Add return type annotation: `None`

(ANN206)